src/test/java/com/desiato/puresynth/services/UserServiceTest.java

```diff
@@ -0,0 +1,33 @@
+package com.desiato.puresynth.services;
+
+import com.desiato.puresynth.BaseTest;
+import com.desiato.puresynth.dtos.AuthenticatedUser;
+import com.desiato.puresynth.models.Session;
+import com.desiato.puresynth.models.User;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class UserServiceTest extends BaseTest {
+
+    @Test
+    @Transactional
```

Comment:
Because otherwise I am getting this error `No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call`.
I tried to invert the sequence of deletions, first the session and then the user, but I still get the error

Reply:
This is a bit related to OSIV: https://www.baeldung.com/spring-open-session-in-view. In the endpoint logic, Spring opens a db connection and transaction. By calling a service method, it does not. Maybe it's why you needed it there.

Also you probably need to enable transaction management - don't remember if you already did it: https://www.baeldung.com/transaction-configuration-with-jpa-and-spring.
Also a good read is on the page: https://docs.spring.io/spring-framework/docs/4.2.x/spring-framework-reference/html/transaction.html

Reply:
Thanks for the links! I removed the Transactional annotation from the test.
I removed UserServiceTest, because its functionality has been replaced by `deleteUser_ShouldDeleteUserAndReturnOk()` in UserControllerTest